Release notes for v0.10.2-alpha (#1054)
spiceai/spiceai: docs/release_notes/v0.10.2-alpha.md

# Spice v0.10.2-alpha (Apr 9, 2024)

> The v0.10.2-alpha release adds support for MySQL and makes external data connections more robust on initialization.

Reviewer: ```suggestion
The v0.10.2-alpha release adds the MySQL data connector and makes external data connections more robust on initialization.
```

## Highlights in v0.10.2-alpha

> - **Connect to MySQL instances**: Connect to any MySQL server, including SSL support.

Reviewer: ```suggestion
- **MySQL data connector**: Connect to any MySQL server, including SSL support.
```

> - **Data connections verified at instantialisation**: Verify endpoints and authorisation for external data connections (e.g. databricks, spice.ai) at initialization. 

Reviewer 2: ```suggestion
- **Data connections verified at instantialisation**: Verify endpoints and authorization for external data connections (e.g. databricks, spice.ai) at initialization. 
```

## New Contributors
* @rthomas made their first contribution in https://github.com/spiceai/spiceai/pull/1022
* @ahirner made their first contribution in https://github.com/spiceai/spiceai/pull/1029
* @gloomweaver made their first contribution in https://github.com/spiceai/spiceai/pull/1004

## Contributors
- @phillipleblanc
- @y-f-u
- @ewgenius
- @sgrebnov
- @lukekim
- @digadeesh
- @jeadie

## New in this release

> - **Adds** Support for MySQL as a dataconnector by @gloomweaver in https://github.com/spiceai/spiceai/pull/1004

Reviewer: ```suggestion
- **Adds** MySQL data connector by @gloomweaver in https://github.com/spiceai/spiceai/pull/1004
```

- **Fixes** `show tables;` parsing in the Spice SQL repl.

> - **Adds** verifying external data connections at initialization

Reviewer: ```suggestion
- **Adds** data connector verification at initialization
```

    - For Dremio by @sgrebnov in https://github.com/spiceai/spiceai/pull/1017
    - For Databricks by @sgrebnov in https://github.com/spiceai/spiceai/pull/1019
    - For Spice.ai by @sgrebnov in https://github.com/spiceai/spiceai/pull/1020

> - **Fixes** Ensures unit and doc tests compile and run. by @rthomas in https://github.com/spiceai/spiceai/pull/1022

Reviewer: ```suggestion
- **Fixes** Ensures unit and doc tests compile and run by @rthomas in https://github.com/spiceai/spiceai/pull/1022
```
